### PR TITLE
ZoneHVAC:PackagedTerminalAirConditioner transition to UnitarySystem

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes DO NOT MERGE
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/UnitarySystem.hh [hunk 1]
```diff
@@ -424,6 +431,24 @@ namespace UnitarySystems {
...
+        Real64 m_CoolConvTol;
+        int m_HVACSizingIndex;
+        int m_AvailStatus;
+        bool m_IsZoneEquipment;
+        bool m_ZoneCompFlag;
+        std::string m_AvailManagerListName;
+        int m_ZoneUnitaryNum;  // 1-based index of this parent type for zone equipment processing
+        int m_PTACNum;         // 1-based index of this parent type for zone equipment processing
+        int m_PTHPNum;         // 1-based index of this parent type for zone equipment processing
+        int m_PTWSHPNum;       // 1-based index of this parent type for zone equipment processing
```
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh]: These 4 (m_ZoneUnitaryNum through m_PTWSHPNum) are for SysAvailManagers. This was my first attempt at storing the relative index to each Eq type. e.g., 12 UnitarySystems of different types but only 3 PTACs, with PTAC index 1, 2 and 3. This concept should be able to be stored in a single variable for all Eq types, ZoneEqCompNum or something like that.

**Comment by rraustad** [comment]
Tested SZVAV on PackagedTermianlAirConditioner with annual simulation. Clean err file. Develop versus this branch with PackagedTerminalHeatPump cc/hh source files removed.

image

**Comment by rraustad** [comment]
I think I will need help figuring this one out. I don't know enough to know if it is due to deleting dataPTHP from state, or if the UnitarySys class is getting too large, or ?? @Myoldmopar @mitchute @anyone, could you run this example file and give some pointers please.

Linux CI shows that example file DOAToUnitarySystem fails with corrupted double-linked list (not small). I ran that file and an exception occurs here:

image

The Linux CI shows:

image

**Comment by rraustad** [comment]
During this migration of packaged terminal models over to UnitarySystem, those models were compared to V9.6 pre-this branch, to aid in the code adjustments. Things actually look remarkably good. There are some component sizing issues in some corner cases, which shows on the CI results yet not in the local testing done to date (i.e., any code updates to get PTUnits to work the same may change existing UnitarySystem results).

So, showing here results of these comparisons (with some modifications if necessary):

 - PackagedTerminalAirConditioner
 - PackagedTerminalHeatPump
 - PackagedTerminalHeatPump_SysAvailManagerList (added to PTHP1)
 - DOAToWaterToAirHPInlet (TBD)

**Comment by rraustad** [comment]
image

**Comment by rraustad** [comment]
image

**Comment by rraustad** [comment]
image

**Comment by rraustad** [comment]
image

**Comment by rraustad** [comment]
@mjwitte I am not sure what a change in the audit diffs represent.
First diff file = ASHRAE901_HotelSmall_STD2019_Denver

    - sizeCompSizeTableEntry=        1600
    - numCompSizeTableEntry=         852
    + sizeCompSizeTableEntry=         800
    + numCompSizeTableEntry=         744

**Comment by rraustad** [comment]
The "corrupted double-linked list" compiler warning turned out to be an Array1D index issue (0 not good), identified by @Myoldmopar.

**Comment by mjwitte** [comment]
> I am not sure what a change in the audit diffs represent.
First diff file = ASHRAE901_HotelSmall_STD2019_Denver

    - sizeCompSizeTableEntry=        1600
    - numCompSizeTableEntry=         852
    + sizeCompSizeTableEntry=         800
    + numCompSizeTableEntry=         744


`numCompSizeTableEntry` counts how many entries there are in the Component Sizing Summary table output. It's incremented in `AddCompSizeTableEntry` which gets called by `reportSizeOutput`. So, the old PTAC must be reporting some additional sizing outputs that unitary does not?

**Comment by nrel-bot-3** [comment]
@rraustad @lgentile it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad @lgentile it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@rraustad @lgentile it has been 28 days since this pull request was last updated.

**Comment by mjwitte** [comment]
Closing this for now, but keeping the branch for reference, because the new branch in #9273 had lots of conflicts.

**Pull request closed without merging**